crates/rpc-client/src/contract.rs

```diff
@@ -0,0 +1,250 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Instant;
+
+use anyhow::{anyhow, bail, Result};
+use arc_swap::{ArcSwap, Guard};
+use async_jsonrpc_client::{HttpClient, Params as ClientParams, Transport};
+use gw_config::{BlockProducerConfig, ContractTypeScriptConfig, ContractsCellDep};
+use gw_jsonrpc_types::blockchain::{CellDep, Script};
+use gw_types::packed::RollupConfig;
+use gw_types::prelude::Pack;
+use serde_json::json;
+
+use crate::indexer_types::{Cell, Order, Pagination, ScriptType, SearchKey};
+use crate::rpc_client::RPCClient;
+use crate::utils::to_result;
+
+// Used in block producer and challenge
+#[derive(Clone)]
+pub struct ContractsCellDepManager {
+    rpc_client: RPCClient,
+    scripts: Arc<ContractTypeScriptConfig>,
+    deps: Arc<ArcSwap<ContractsCellDep>>,
+}
+
+impl ContractsCellDepManager {
+    pub async fn build(rpc_client: RPCClient, scripts: ContractTypeScriptConfig) -> Result<Self> {
+        let now = Instant::now();
+        let deps = query_cell_deps(&rpc_client, &scripts).await?;
+        log::trace!("[contracts dep] build {}ms", now.elapsed().as_millis());
+
+        Ok(Self {
+            rpc_client,
+            scripts: Arc::new(scripts),
+            deps: Arc::new(ArcSwap::new(Arc::new(deps))),
```

Comment:
This line makes me think for 20min at least... I suspect both "Arc"s are not necessary. 
I guess we can just use a normal pointer here guarded by ArcSwap.
```rust
deps: ArcSwap::from_pointee(deps);
```

Reply:
Done